### PR TITLE
Reduce SEIR startup costs

### PR DESCRIPTION
**Pull request by samshah**
The biggest cost in a single sim SEIR run was importation of Numba and the JIT compilation. Change this to compile ahead of time, which results in a nice 60% lift (13.4s -> 5.1s) in one run SEIR performance by saving these startup costs---which will be valuable for our large inference jobs.

Minor performance benefit when running many simulations as JIT costs are amortized away.

```
Benchmark #1: single sim JIT compilation (current)
  Time (mean ± σ):     13.429 s ±  0.537 s
  Range (min … max):   12.973 s … 14.867 s    100 runs

Benchmark #2: single sim AOT compilation (new)
  Time (mean ± σ):      5.129 s ±  0.125 s
  Range (min … max):    4.901 s …  5.364 s    100 runs
```

### CODE REVIEW

**Comment by samshah** [comment]
For now, you'll need to run `python setup.py install` (1st time) and then can run `simulate.py` or `python -m SEIR` to start the simulation. I'll look into making this easier once the other Docker updates are in.

**Comment by kkintaro** [comment]
Recommend adding build directory to .gitignore

File: R/scripts/make_makefile.R [hunk 1]
```diff
@@ -124,7 +124,7 @@ simulation_make_command <- function(simulation,scenario,previous_simulation, pre
   if(using_static_filter){
     dependency_name <- paste(dependency_name, filter_target_name(simulation,prefix))
   }
-  command_name <- paste0("$(PYTHON) $(PIPELINE)/simulate.py -c $(CONFIG) -s ",scenario," -n ",simulation - previous_simulation," -j $(NCOREPER)")
+  command_name <- paste0("$(PYTHON) -m SEIR -c $(CONFIG) -s ",scenario," -n ",simulation - previous_simulation," -j $(NCOREPER)")
```
Comment (kkintaro) [inline R/scripts/make_makefile.R:127]: Are "-m SEIR" and "simulate.py" interchangeable?

Comment (samshah) [reply R/scripts/make_makefile.R:127]: correct, they run the same code

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,6 +24,9 @@ jobs:
       - name: Set up Rprofile
         run: cp Docker.Rprofile $HOME/.Rprofile
         shell: bash
+      - name: Install SEIR package
+        run: /home/app/python_venv/bin/python setup.py install
+        shell: bash
```
Comment (kkintaro) [inline .github/workflows/ci.yml:29]: Not necessary for this PR, but can this be combined with the local_install.R?

Comment (samshah) [reply .github/workflows/ci.yml:29]: what do you mean combined? i can have them run in 1 CI step with `&&` but would rather work on a solution that obviates the need for `local_install.R`

**Comment by samshah** [comment]
**n.b.** It'll be difficult to get this to run outside of the container on non-Unix systems (e.g., raw Windows) because you'll need to have the ability to compile and link C code.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kkintaro) [reply .github/workflows/ci.yml:29]: I was thinking of combining into one script, but getting of local_install.R later on would be much, much better.

Comment (jkamins7) [reply .github/workflows/ci.yml:29]: local_install.R is not necessary for the docker now, so i think removing it from ci should be straightforward

**Comment by jkamins7** [comment]
hm... not working on windows might be hard, since we have at least one person on windows unable to use docker...  this change makes it impossible to run without the install?

**Comment by samshah** [comment]
there isn't a way to run w/o compiling the module. you _can _compile it on windows, it's no different than compiling any other python module, but there's a chance there'll be some environment pain.

**Pull request merged**